### PR TITLE
[#164772597] Add soft logout method

### PR DESCRIPTION
**Pull request by lmorelli986**
This PR aims to split logout in two possible actions:
* logout keeping user data
* logout resetting user data

### Android
screen

### iOS
screen2

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#164772597](https://www.pivotaltracker.com/story/show/164772597): Quando effettuo il logout e poi nuovamente login nell'app, lo stato di lettura dei messaggi e la loro presenza in archivio non vengono conservati

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/store/actions/authentication.ts [hunk 1]
```diff
@@ -12,6 +12,10 @@ import { PublicSession } from "../../../definitions/backend/PublicSession";
 import { IdentityProvider } from "../../models/IdentityProvider";
 import { SessionToken } from "../../types/SessionToken";
 
+export type logoutMethod = {
```
Comment (francescopersico) [inline ts/store/actions/authentication.ts]:
```suggestion
export type LogoutMethod = {
```

File: ts/components/SelectLogoutMethod.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+});
+
+type Props = {
+  onCancel: () => void;
+  onMethodSelected: (_: logoutMethod) => void;
+  header?: React.ReactNode;
+};
+
+/**
+ * A modal that allow the user to select one of the device available Calendars
```
Comment (francescopersico) [inline ts/components/SelectLogoutMethod.tsx]: No Calendars here :)

File: ts/screens/profile/ProfileMainScreen.tsx [hunk 3]
```diff
@@ -76,11 +85,30 @@ class ProfileMainScreen extends React.PureComponent<Props> {
     Toast.show({ text: "The cache has been cleared." });
   };
 
+  private logout = (keepUserData: logoutMethod) => {
```
Comment (francescopersico) [inline ts/screens/profile/ProfileMainScreen.tsx]: why keepUserData?

Comment (lmorelli986) [reply ts/screens/profile/ProfileMainScreen.tsx]: Right, I forgot tu update it as well. Changed.

File: ts/components/SelectLogoutMethod.tsx (hunk 2 above)
Comment (lmorelli986) [reply ts/components/SelectLogoutMethod.tsx]: Thx, changed.

File: ts/store/actions/authentication.ts (hunk 1 above)
Comment (lmorelli986) [reply ts/store/actions/authentication.ts]: Fixed.

File: locales/en/index.yml [hunk 4]
```diff
@@ -57,7 +57,16 @@ profile:
   sections:
     spid: SPID PROFILE
   unknown: UNKNOWN
-  logout: Logout from this SPID identity
+  logout:
+    menulabel: Logout from this SPID identity
+    cta: 
+      header: How do you want to quit the app?
```
Comment (cloudify) [inline locales/en/index.yml]: this is quite confusing :)

File: locales/it/index.yml [hunk 5]
```diff
@@ -58,7 +58,16 @@ profile:
   sections:
     spid: PROFILO SPID
   unknown: SCONOSCIUTO
-  logout: Esci da questa identità SPID
+  logout: 
+    menulabel: Esci da questa identità SPID
+    cta: 
+      header: Come vuoi uscire dall'app?
```
Comment (cloudify) [inline locales/it/index.yml]: @matteodesanti can you review the copy here? it looks quite confusing

File: ts/components/SelectLogoutMethod.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  private onBackPress = () => {
+    this.props.onCancel();
+    // Returning true is mandatory to avoid the default press action to be
+    // triggered as if the modal was not visible
+    return true;
+  };
+
+  public render() {
+    return (
+      <Container>
```
Comment (cloudify) [inline ts/components/SelectLogoutMethod.tsx]: is this a screen? then it should be named `SelectLogoutMethod` and should be under `screens`

File: ts/components/SelectLogoutMethod.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+        <Content style={styles.content}>
+          <React.Fragment>{this.props.header || null}</React.Fragment>
```
Comment (cloudify) [inline ts/components/SelectLogoutMethod.tsx]: is `React.Fragment` needed?

File: ts/components/SelectLogoutMethod.tsx (hunk 6 above)
Comment (lmorelli986) [reply ts/components/SelectLogoutMethod.tsx]: It is a modal, like _SelectCalendarModal_. Would rather _SelectLogoutMethodModal_ be a more suitable name?

File: ts/components/SelectLogoutMethod.tsx (hunk 7 above)
Comment (lmorelli986) [reply ts/components/SelectLogoutMethod.tsx]: Not, really. Changed.

File: ts/components/SelectLogoutMethod.tsx (hunk 6 above)
Comment (cloudify) [reply ts/components/SelectLogoutMethod.tsx]: yes, thanks

File: ts/store/actions/authentication.ts [hunk 8]
```diff
@@ -12,6 +12,10 @@ import { PublicSession } from "../../../definitions/backend/PublicSession";
 import { IdentityProvider } from "../../models/IdentityProvider";
 import { SessionToken } from "../../types/SessionToken";
 
+export type LogoutMethod = {
```
Comment (fpersico) [inline ts/store/actions/authentication.ts]: I think LogoutOptions is better name.

File: ts/components/SelectLogoutMethod.tsx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+  public componentDidMount() {
+    BackHandler.addEventListener("hardwareBackPress", this.onBackPress);
+  }
+
+  public componentWillUnmount() {
+    BackHandler.removeEventListener("hardwareBackPress", this.onBackPress);
+  }
+}
+
+export default connect()(SelectLogoutMethod);
```
[the remark this answers is not in the document]
Comment (lmorelli986) [reply ts/components/SelectLogoutMethod.tsx]: removed.

File: ts/store/actions/authentication.ts (hunk 8 above)
Comment (lmorelli986) [reply ts/store/actions/authentication.ts]: changed

**Pull request merged**